app/models/virtual_cap_pool.rb

```diff
@@ -0,0 +1,41 @@
+class VirtualCapPool < ApplicationRecord
+  belongs_to :responsible_body
+  has_many :school_virtual_caps, dependent: :destroy
+  has_many :school_device_allocations, through: :school_virtual_caps
+  has_many :schools, through: :school_device_allocations
+
+  enum device_type: {
+    'coms_device': 'coms_device',
+    'std_device': 'std_device',
+  }
+
+  def recalculate_caps!
+    self.cap = school_device_allocations.sum(:cap)
+    self.devices_ordered = school_device_allocations.sum(:devices_ordered)
+    save!
+    # TODO: trigger events for CC here?
+  end
+
+  def add_school(school)
```

Comment:
Did you imagine the caller of this would check that this is an eligible school? Or perhaps it belongs here?

Reply:
I've added some additional checks here to ensure the school is in a `can_order` or `can_order_for_specific_circumstances` state when adding to the pool, but this logic/checks might better be handled by whatever it is that puts the school in that state.